packages/@lwc/engine/src/framework/template.ts

```diff
@@ -47,8 +45,9 @@ export function isVMBeingRendered(vm: VM) {
 }
 
 export { registerTemplate };
+
 export interface Template {
-    (api: RenderAPI, cmp: object, slotSet: SlotSet, ctx: Context): VNodes;
+    (api: RenderAPI, cmp: object, slotSet: SlotSet, ctx: Record<string, any>): VNodes;
```

Comment:
What is the difference between using type `: object` v/s `: Record<string, any>`?

Reply:
The `object` type represent anything that is [not a primitive type](https://www.typescriptlang.org/docs/handbook/basic-types.html#object). This type is really confusing and we should really think twice before using it:

```ts
const a: object = {} // valid
const b: object = function() {} // also valid
const c: object = new Map() // also valid 
```

The `Record<K, T>` is a utility type equivalent to `{ [P in K]: T }`. There is a really interesting discussion about why and when this utility type should be used on [stack overflow](https://stackoverflow.com/questions/51936369/what-is-the-record-type-in-typescript). The gist of it is it the `Record` type can be used to restrict the set of keys available, for example: `Record<'connectedCallback' | 'disconnectedCallback' | 'renderedCallback', () => {}>`. After ready the thread, it looks like the more idiomatic approach is to use `{ [key: string]: any }` to type context.